dbgpt/storage/knowledge_graph/community_summary.py

```diff
@@ -0,0 +1,168 @@
+        return [chunk.chunk_id for chunk in chunks]
+
+    async def asimilar_search_with_scores(
+        self,
+        text,
+        topk,
+        score_threshold: float,
+        filters: Optional[MetadataFilters] = None,
+    ) -> List[Chunk]:
+
+        # Determine if search is global or local
+        is_global_search = await self._get_intent_from_query(text)
+        return await (self._global_search if is_global_search else self._local_search)(
+            text, topk, score_threshold, filters
+        )
+
+    async def _global_search(
+        self,
+        text,
+        topk,
+        score_threshold: float,
+        filters: Optional[MetadataFilters] = None,
+    ) -> List[Chunk]:
+        community_summaries = await self.community_store.summarize_communities()
+        chunks = [
+            Chunk(
+                content=await self._generate_answer_from_summary(summary, text),
+                metadata={"cluster_id": cluster_id},
+            )
+            for cluster_id, summary in community_summaries.items()
+        ]
+        return chunks[:topk]
+
+        #     final_answer = self._aggregate_answers(community_summaries)
+        # return final_answer
+
+    async def _local_search(
+        self,
+        text,
+        topk,
+        score_threshold: float,
+        filters: Optional[MetadataFilters] = None,
+    ) -> List[Chunk]:
+        """Search neighbours on knowledge graph."""
+        if not filters:
+            logger.info("Filters on knowledge graph not supported yet")
+
+        # extract keywords and explore graph store
+        keywords = await self._keyword_extractor.extract(text)
+        subgraph = self._graph_store.explore(keywords, limit=topk)
+        logger.info(f"Search subgraph from {len(keywords)} keywords")
+
+        content = (
+            "The following vertices and edges data after [Subgraph Data] "
+            "are retrieved from the knowledge graph based on the keywords:\n"
+            f"Keywords:\n{','.join(keywords)}\n"
+            "---------------------\n"
+            "You can refer to the sample vertices and edges to understand "
+            "the real knowledge graph data provided by [Subgraph Data].\n"
+            "Sample vertices:\n"
+            "(alice)\n"
+            "(bob:{age:28})\n"
+            '(carry:{age:18;role:"teacher"})\n\n'
+            "Sample edges:\n"
+            "(alice)-[reward]->(alice)\n"
+            '(alice)-[notify:{method:"email"}]->'
+            '(carry:{age:18;role:"teacher"})\n'
+            '(bob:{age:28})-[teach:{course:"math";hour:180}]->(alice)\n'
+            "---------------------\n"
+            f"Subgraph Data:\n{subgraph.format()}\n"
+        )
+        return [Chunk(content=content, metadata=subgraph.schema())]
+
+    async def _generate_answer_from_summary(self, community_summary, query):
+        """Generate an answer from a community summary based on a given query using LLM."""
+        response = client.chat.completions.create(
```

Comment:
@M1n9X 
Invoking the LLM service should not directly use the SDK of OpenAI. Refer to the implementation of `LLMExtractor`.